payments/models.py

```diff
@@ -957,3 +962,56 @@ class OrderToken(UUIDModel, TimeStampedModel):
     @property
     def is_valid(self):
         return not self.cancelled and now() < self.valid_until
+
+
+class AbstractOffer(UUIDModel, TimeStampedModel):
+    customer = models.ForeignKey(
+        CustomerProfile, related_name="offers", on_delete=models.CASCADE
+    )
+    status = models.CharField(
+        choices=OfferStatus.choices, default=OfferStatus.PENDING, max_length=8
+    )
+    due_date = models.DateField(verbose_name=_("due date"), default=default_due_date)
+    # Optional fields to contact
+    customer_first_name = models.TextField(blank=True, null=True)
+    customer_last_name = models.TextField(blank=True, null=True)
+    customer_email = models.TextField(blank=True, null=True)
+    customer_phone = models.TextField(blank=True, null=True)
+
+    class Meta:
+        abstract = True
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
+        super().save(*args, **kwargs)
+
+    def update_from_profile(self, profile_token: str):
+        profile = ProfileService(profile_token).get_profile(self.customer.id)
+
+        self.customer_first_name = profile.first_name
+        self.customer_last_name = profile.last_name
+        self.customer_email = profile.email
+        self.customer_phone = profile.phone
+
+        self.save()
+
+
+class BerthSwitchOffer(AbstractOffer):
+    application = models.ForeignKey(
+        BerthApplication, related_name="switch_offers", on_delete=models.CASCADE
+    )
+    lease = models.ForeignKey(
```

Comment:
Could add validation that this lease must be for the season that immediately precedes the season defined in the application?
Why linking to lease instead of Order?

Reply:
> _Could add validation that this lease must be for the season that immediately precedes the season defined in the application?_

Good catch 👍 

> _Why linking to lease instead of Order?_

It's a link to the _old_ lease, since the lease is kinda the "source of truth", meaning that even if the invoice is cancelled or anything, as long as the lease is not terminated, it's what determines whether the customer has the place or not.